src/main/java/com/darksci/pardot/api/response/tagobject/TagObject.java

```diff
@@ -39,10 +40,23 @@ public Long getTagId() {
         return tagId;
     }
 
-    public String getType() {
+    /**
+     * @return Type as returned by the API in string format.
+     */
+    public String getTypeName() {
         return type;
     }
 
+    /**
+     * @return Enum representing the type.
+     */
+    public TagObjectType getType() {
```

Comment:
getType() now returns the Enum.  To keep these future proof I added getTypeName() to return the value returned from the Api in case new types get added in the future.